ecl/hqlcpp/hqlsource.cpp

```diff
@@ -1135,14 +1137,17 @@ void SourceBuilder::rebindFilepositons(BuildCtx & ctx, IHqlExpression * dataset,
     {
         OwnedHqlExpr selector = createSelector(side, dataset, selSeq);
         OwnedHqlExpr selectorFpos = getFilepos(selector, isLocal);
-        ctx.associateExpr(selectorFpos, match->queryExpr());
+        ctx.associateOwn(*new MonitoredDefinedValue(transformUsesVirtualFilePosition, selectorFpos, match->queryExpr()));
     }
 }
 
 
 void SourceBuilder::rebindFilepositons(BuildCtx & ctx, IHqlExpression * dataset, node_operator side, IHqlExpression * selSeq)
 {
-    bool savedIsVirtualLogicalFilenameUsed = isVirtualLogicalFilenameUsed;  // don't allow the rebinding to set the flag.
+    // don't allow the rebinding to modify these flags.
```

Comment:
This code looks a bit iffy. Wouldn't just passing a dummy local bool on line 1159 have the same effect?

Reply:
No, that does need be the correct value.  The problem is that the act of looking up what the fileposition maps to sets the flag - but we only want it to be set if the generated code looks up the fileposition.